superset/sql_parse.py

```diff
@@ -182,7 +182,7 @@ def _extract_limit_from_query(self, statement):
             _, token = statement.token_next(idx=idx)
             if token:
                 if isinstance(token, IdentifierList):
-                    _, token = token.token_next(idx=-1)
+                    token = token.tokens[-1]
```

Comment:
Could you use `token_next(..., reverse=True)`. This skips comments, whitespace etc. and thus is probably more robust.

Reply:
I agree using the native methods would be preferable, and I actually tried to go down that path, but it seems the reverse argument is in fact private (`_reverse`), and didn't seem to work similar to `sorted(..., reverse=True)`. Will see if I can come up with a more robust solution.

Reply:
Can you check my new proposal @john-bodley ?